jcl/src/jdk.jcmd/share/classes/openj9/tools/attach/diagnostics/tools/Jmap.java

```diff
@@ -0,0 +1,156 @@
+/*[INCLUDE-IF Sidecar18-SE]*/
+/*******************************************************************************
+ * Copyright (c) 2019, 2019 IBM Corp. and others
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+ * or the Apache License, Version 2.0 which accompanies this distribution and
+ * is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following
+ * Secondary Licenses when the conditions for such availability set
+ * forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+
+package openj9.tools.attach.diagnostics.tools;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Properties;
+import com.ibm.tools.attach.target.AttachHandler;
+import com.ibm.tools.attach.target.IPC;
+import openj9.tools.attach.diagnostics.attacher.AttacherDiagnosticsProvider;
+import openj9.tools.attach.diagnostics.base.DiagnosticProperties;
+import openj9.tools.attach.diagnostics.base.DiagnosticUtils;
+
+/**
+ * JMap A tool for listing heap information about another Java process
+ *
+ */
+public class Jmap {
+
+	private static List<String> vmids;
+	private static boolean histo;
+	private static boolean live;
+	@SuppressWarnings("nls")
+	private static String HELPTEXT = "jmap: obtain heap information about a Java process%n"
+			+ " Usage:%n"
+			+ "    jmap <option>* <vmid>%n"
+			+ "        <vmid>: Attach API VM ID as shown in jps or other Attach API-based tools%n"
+			+ "        <vmid>s are read from stdin if none are supplied as arguments%n"
+			+ "    -histo: print statistics about classes on the heap, including number of objects and aggregate size%n"
+			+ "    -histo:live : Print only live objects%n"
+			+ "    -J: supply arguments to the Java VM running jps%n"
+			+ " NOTE: this utility may significantly affect the performance of the target JVM.%n"
+			+ "At least one option must be selected.";
```

Comment:
The final `%n` is missing.

Reply:
The terminating `%n` is added by the `printf` ca. line 148. Would you prefer that it move into the string proper?